pkg/kubectl/cmd/diff.go

```diff
@@ -417,25 +453,29 @@ func RunDiff(f cmdutil.Factory, diff *DiffProgram, options *DiffOptions, from, t
 		Unstructured().
 		NamespaceParam(cmdNamespace).DefaultNamespace().
 		FilenameParam(enforceNamespace, &options.FilenameOptions).
+		Local().
 		Flatten().
 		Do()
 	if err := r.Err(); err != nil {
 		return err
 	}
 
+	dl, err := NewDownloader(f)
+	if err != nil {
+		return err
+	}
+
 	err = r.Visit(func(info *resource.Info, err error) error {
 		if err != nil {
 			return err
 		}
 
-		if err := info.Get(); err != nil {
-			if !errors.IsNotFound(err) {
-				return cmdutil.AddSourceToErr(fmt.Sprintf("retrieving current configuration of:\n%s\nfrom server for:", info.String()), info.Source, err)
-			}
-			info.Object = nil
+		remote, err := dl.Download(info)
+		if err != nil {
+			glog.Warningf("failed to download remote object: %v", err)
```

Comment:
Nit but human warnings aren't via glog, we use `fmt.Fprintf(stdErr, "warning: ...")`

glog is terrible for humans.